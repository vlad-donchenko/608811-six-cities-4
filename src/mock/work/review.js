const reviews = [
  {
    id: 0,
    date: `2019-05-08T14:13:56.569Z`,
    name: `Max`,
    rating: 4,
    avatar: `avatar-max.jpg`,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  },
  {
    id: 1,
    date: `2019-05-08T14:13:56.569Z`,
    name: `Angelina`,
    rating: 2,
    avatar: `avatar-max.jpg`,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century. Quiet cozy and picturesque that hides behind a a river`
  },
  {
    id: 2,
    date: `2019-05-08T14:13:56.569Z`,
    name: `Andrew`,
    rating: 5,
    avatar: `avatar-max.jpg`,
    text: `The building is green and from 18th century. Quiet cozy and picturesque that hides behind a a river`
  },
  {
    id: 3,
    date: `2019-05-08T14:13:56.569Z`,
    name: `Andrew`,
    rating: 1,
    avatar: `avatar-max.jpg`,
    text: `Quiet cozy and picturesque that hides behind a a river`
  },
  {
    id: 4,
    date: `2019-05-08T14:13:56.569Z`,
    name: `Jim`,
    rating: 5,
    avatar: `avatar-max.jpg`,
    text: `Quiet cozy and picturesque that hides behind a a river. The building is green and from 18th century. Quiet cozy and picturesque that hides behind a a rive`
  }
];

export {reviews};
