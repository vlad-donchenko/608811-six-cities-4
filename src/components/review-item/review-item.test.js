import React from "react";
import renderer from 'react-test-renderer';
import ReviewItem from "./review-item.jsx";
import {reviews} from "../../mock/test/review";

it(`Should render ReviewsList`, () => {
  const tree = renderer.create(<ReviewItem review={reviews[0]}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
