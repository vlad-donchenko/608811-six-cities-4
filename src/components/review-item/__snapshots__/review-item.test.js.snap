// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ReviewsList 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Jim"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Jim
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-8"
        >
          June 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Nick"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Nick
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Quiet cozy and picturesque that hides behind a a river
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-8"
        >
          May 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Andrew"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Andrew
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The building is green and from 18th century. Quiet cozy and picturesque that hides behind a a river
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-8"
        >
          May 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Max"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Quiet cozy and picturesque that hides behind a a river
        </p>
        <time
          className="reviews__time"
          dateTime="2019-06-8"
        >
          July 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Jim"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Jim
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Quiet cozy and picturesque that hides behind a a river. The building is green and from 18th century. Quiet cozy and picturesque that hides behind a a rive
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-8"
        >
          May 2019
        </time>
      </div>
    </li>
  </ul>
</section>
`;
